crates/hotshot/src/traits/networking/combined_network.rs

```diff
@@ -139,6 +162,105 @@ impl<TYPES: NodeType> CombinedNetworks<TYPES> {
     pub fn secondary(&self) -> &Libp2pNetwork<Message<TYPES>, TYPES::SignatureKey> {
         &self.networks.1
     }
+
+    /// a helper function returning a bool whether a given message is of delayable type
+    fn should_delay(message: &Message<TYPES>) -> bool {
+        match &message.kind {
+            MessageKind::Consensus(consensus_message) => match &consensus_message.0 {
+                Either::Left(general_consensus_message) => {
+                    matches!(general_consensus_message, GeneralConsensusMessage::Vote(_))
+                }
+                Either::Right(_) => true,
+            },
+            MessageKind::Data(_) => false,
+        }
+    }
+
+    /// a helper function to send messages through both networks (possibly delayed)
+    async fn send_both_networks(
+        &self,
+        message: Message<TYPES>,
+        recipients: BTreeSet<TYPES::SignatureKey>,
+        message_type: MessageType,
+    ) -> Result<(), NetworkError> {
```

Comment:
I wonder if instead of adding this `MessageType` enum and having match here, you could pass this function two futures called `primary` and `backup`. 
maybe
```rust
async fn send_both_networks(
    &self, primary: impl Future<Output = Result<(), NetworkError>>, 
    backup: impl Future<Output = Result<(), NetworkError>>)
```

Then the call would be like:
```rust
self.send_both_networks(
    self.primary().broadcast_message(message.clone(), recipients.clone(), 
    self.secondary().broadcast_message(...)).await
```

Reply:
Not 100% sure this works, but something like that should